Extend `separate-draft-pr-button` to `one-click-pr-or-gist`

darkred opened the pull request:
<!--

Thanks for contributing! 🍄 Do not ignore this template, plz.

1. Does this PR close/fix an existing issue? Write something like `Closes #10`

Help us test and visualize this PR:

2. What pages does this PR affect? Include some REAL URLs where you tested the code
3. If applicable, add demonstrative screenshots or gifs

Lastly:

4. Open the PR as draft and review it yourself first. Fix what you find and explain weird code, if necessary.

-->

Closes #4366.
Follows https://github.com/sindresorhus/refined-github/issues/4366#issuecomment-846079038

## Test URLs
https://gist.github.com/

## Screenshot
![](https://user-images.githubusercontent.com/723651/119201621-f64cfa00-ba97-11eb-9fe1-a4fa28ffa778.jpg)

fregante commented:
@yakov116’s suggestion was to just change the selector and tweak the CSS instead of copy-pasting the whole code

darkred commented:
*I wrongly thought that apart from his suggestion, something like this was also needed:*
`title = (/\bsecret\b/i.test(title)) ? 'Create secret Gist' : 'Create public Gist';` 
*Sorry about that.*

---

So, I removed the extra feature files, 
and just renamed the existing `separate-draft-pr-button` to `separate-draft-pr-gist-creation` (and added a .css file with the same name).

darkred commented:
Also, @fregante, should I join (vertically) the [existing screenshot](https://user-images.githubusercontent.com/202916/67269317-cd791300-f4b6-11e9-89d1-392de7ef71e1.png) of the feature,
with the [one](https://user-images.githubusercontent.com/723651/119201621-f64cfa00-ba97-11eb-9fe1-a4fa28ffa778.jpg) I have in OP, for gists ?

fregante commented:
If you join them, the new screenshot should be composed exactly the same way, on the right side. If the style doesn’t match it’s just confusing

darkred commented:
Ok, done: 

<details>
<summary>New screeshot</summary>

![](https://user-images.githubusercontent.com/723651/119255867-9ef97780-bbc6-11eb-8588-ce66c6478d5d.jpg)
</details>

fregante commented on readme.md:
```diff
@@ -269,7 +269,7 @@ Thanks for contributing! 🦋🙌
 - [](# "pull-request-hotkey") [Adds keyboard shortcuts to cycle through PR tabs: <kbd>g</kbd> <kbd>←</kbd> and <kbd>g</kbd> <kbd>→</kbd>, or <kbd>g</kbd> <kbd>1</kbd>, <kbd>g</kbd> <kbd>2</kbd>, <kbd>g</kbd> <kbd>3</kbd> and <kbd>g</kbd> <kbd>4</kbd>](https://user-images.githubusercontent.com/16872793/94634958-7e7b5680-029f-11eb-82ea-1f96cd11e4cd.png).
 - [](# "pr-branch-auto-delete") Automatically deletes the branch right after merging a PR, if possible.
-- [](# "separate-draft-pr-button") [Lets you create draft pull requests in one click.](https://user-images.githubusercontent.com/202916/67269317-cd791300-f4b6-11e9-89d1-392de7ef71e1.png)
+- [](# "separate-draft-pr-gist-creation-button") [Lets you create draft pull requests and public/secret gists in one click.](https://user-images.githubusercontent.com/723651/119255867-9ef97780-bbc6-11eb-8588-ce66c6478d5d.jpg).
```
This feature is similar to `unwrap-useless-dropdowns`, except the dropdown isn't "useless". Maybe this name works better:

```suggestion
- [](# "one-click-pr-or-gist") [Lets you create draft pull requests and public gists in one click.](https://user-images.githubusercontent.com/723651/119255867-9ef97780-bbc6-11eb-8588-ce66c6478d5d.jpg).
```

Also secret gists were already one-click

fregante commented on source/features/separate-draft-pr-gist-creation-button.css:
```diff
@@ -0,0 +1,5 @@
+/* For the 1st button "Create public Gist" */
+button[value='1'] {
```
This selector is too broad

darkred replied:
Yes, I thought so.
This `#new_gist > div > #gists+div >button[value="1"]` is good, right?

fregante replied:
One ID is good enough. If nothing better exists: `#gists + div button[value="1"]`

darkred replied:
I used `#gists + div button[value="1"]`. Thank you.

darkred replied:
Btw, there's a think I wanted to report some time now:
when running `npm run lint` I get:

```
Error: No files matching the pattern "'source/**/*.css'" were found.
    at C:\Users\Kostas\Documents\GitHub\refined-github\node_modules\stylelint\lib\standalone.js:212:12
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
ERROR: "lint:css" exited with 1.
```

Maybe the line `"lint:css": "stylelint 'source/**/*.css'",` in `package.json` should change somehow (its glob pattern?) ?
I am asking because in this case, the double-quotes in the selector above would trigger the `string-quotes / Expected single quotes` stylelint rule, but instead I get the error message.

yakov116 replied:
https://github.com/sindresorhus/refined-github/pull/4386

yakov116 commented:
Please add a migration.

darkred commented:
@yakov116 I hope this is what you meant: 
https://github.com/darkred/refined-github/blob/16bd5c23e27b4ac837d6de30bea821a081ffeeb1/source/options-storage.ts#L23

yakov116 commented on source/features/one-click-pr-or-gist.tsx:
```diff
@@ -1,11 +1,12 @@
-	const draftPROption = select('.new-pr-form [name="draft"]');
+	const draftPROption = select('.new-pr-form [name="draft"], #new_gist [name="gist[public]"]');
 	if (!draftPROption) {
 		// 1. Free accounts can't open Draft PRs in private repos, so this element is missing
```
Is the above line still true?

darkred replied:
Yes, but now that I tested it in a draft PR page (I thought it was ok, based on your [comment](https://github.com/sindresorhus/refined-github/issues/4366#issuecomment-846079038)), 
I see now I have to convert
https://github.com/darkred/refined-github/blob/16bd5c23e27b4ac837d6de30bea821a081ffeeb1/source/features/one-click-pr-or-gist.tsx#L24-L28 

to 

```js
if (/\bdraft\b/i.test(title)) {
	title = 'Create draft PR';
}
classList.push('btn-primary');
```

otherwise in PR page, only the button to the right becomes green:
![2021-05-23_174453](https://user-images.githubusercontent.com/723651/119265270-bf3d2c80-bbee-11eb-8299-dad00d7fd1d2.jpg)

&nbsp;
Am I missing something, or should I make the change?

yakov116 replied:
Its fine I don't think it needs to be a different color

darkred replied:
Ok, thank you.

fregante replied:
Checked, yes

yakov116 reviewed (approved):
LGTM

fregante commented:
Thank you for this PR @darkred! 🍦
